internal/ad/mock/libkrb5_mock.c

```diff
@@ -0,0 +1,31 @@
+#include "libkrb5_mock.h"
+
+#include <krb5.h>
+#include <stdio.h>
+#include <string.h>
+
+char *get_krb5_mock_behavior() { return getenv("ADSYS_KRB5_BEHAVIOR"); }
+
+const char *KRB5_CALLCONV krb5_cc_default_name(krb5_context context) {
+    char *behavior = get_krb5_mock_behavior();
+    if (behavior == NULL) {
+        printf("ADSYS_KRB5_BEHAVIOR not set, returning dummy value...");
+        return "FILE:/tmp/krb5cc_0";
+    }
+
+    if (strcmp(behavior, "return_empty_ccache") == 0) {
+        return "";
+    } else if (strcmp(behavior, "return_null_ccache") == 0) {
+        return NULL;
+    } else if (strstr(behavior, "return_ccache") != NULL) {
+        // split the string by the first colon
+        char *ccname = strtok(behavior, ":");
+        ccname = strtok(NULL, "");
+        return ccname;
+    } else if (strstr(behavior, "return_memory_ccache") != NULL) {
+        return "MEMORY:foo";
+    }
```

Comment:
nitpick: Since it's returning on every `if` block, we could remove the `else if` statements and turn this into 4 different `if`s. IMHO, it could improve readability.